src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArr = [];
+  const clone = Object.assign({}, state);
+
+  for (const action in actions) {
+    const actionType = actions[action].type;
+    const removeArr = actions[action].keysToRemove;
+    const extraData = actions[action].extraData;
+
+    if (actionType === 'addProperties') {
+      Object.assign(clone, extraData);
+    } else if (actionType === 'removeProperties') {
+      for (const removeProp of removeArr) {
+        delete clone[removeProp];
+      }
+    } else if (actionType === 'clear') {
+      for (const key in clone) {
+        delete clone[key];
+      }
+    }
```

Comment:
avoid such nesting `else if` conditions
This code is hard to read and maintain
use switch here

Reply:
@Bohdan-Maksymiuk done